gateway/server_test.go

```diff
@@ -216,3 +220,144 @@ func TestGateway_SyncResourcesWithReload(t *testing.T) {
 	})
 
 }
+
+func TestGateway_getHostDetails(t *testing.T) {
+	type checkFn func(*testing.T, *il.BufferedLogger, *Gateway)
+
+	var (
+		orig_readPIDFromFile = readPIDFromFile
+		orig_mainLog         = mainLog
+		orig_getIpAddress    = netutil.GetIpAddress
+		bl                   = il.NewBufferingLogger()
```

Comment:
Should the buffering logger just be a test utility? In that case I misled you, and the log utilities should live in `/test/log.go`. I apologize for the oversight.